lib_collections/models.py

```diff
@@ -495,6 +504,211 @@ def __init__(self, *args, **kwargs):
+            '')
+        related_exhibits = lazy_dotchain(
+            lambda: self.exhibit_page_related_collection.all(),
+            '')
+        collections_by_subject = lazy_dotchain(
+            lambda: self.collection_subject_placements.all(),
+            '')
+        collections_by_format = lazy_dotchain(
+            lambda: self.collection_placements.all(),
+            '')
+
+        output['staff_title'] = staff_title
+        output['staff_position_title'] = staff_position_title
+        output['staff_email'] = staff_email
+        output['staff_phone_number'] = staff_phone_number
+        output['staff_faculty_exchange'] = staff_faculty_exchange
+        output['staff_url'] = staff_url
+
+        output['unit_title'] = unit_title
+        output['unit_url'] = unit_url
+        output['unit_email_label'] = unit_email_label
+        output['unit_phone_label'] = unit_phone_label
+        output['unit_phone_number'] = unit_phone_number
+        output['unit_fax_number'] = unit_fax_number
+        output['unit_link_text'] = unit_link_text
+        output['unit_link_external'] = unit_link_external
+        output['unit_link_page'] = unit_link_page
+        output['unit_link_document'] = unit_link_document
+
+        output["access_location"] = access_location
+
+        output["related_collections"] = related_collections
+        output["collections_by_subject"] = collections_by_subject
+        output["related_exhibits"] = related_exhibits
+        output["collections_by_format"] = collections_by_format
+
+        return output
+
+    def build_breadcrumbs(request):
+        """
+        Create breadcrumb trail.
+
+        Args:
+            HTTP request
+
+        Returns: 
+            Tuple containing breadcrumb trail, along with the
+            final breadcrumb text (which is different because it isn't
+            a link)
+        """
+        breadcrumbs = list(
+            filter(lambda x: x is not "", request.path.split('/'))
+        )
+
+        trimmed_crumbs = breadcrumbs[2:-1]
+        final_crumb = breadcrumbs[-1]
+
+        unslugify_browse = DisplayBrowse.unslugify_browse
+
+        def path_up_to(idx, lst):
+            return {unslugify_browse(lst[i-1]): ("/collex/collections/"
+                                                 + "/".join(lst[:i]))
+                    for i in range(1, idx+1)
+                    if lst[i-1] not in ['list-browse',
+                                        'object',
+                                        'cluster-browse',
+                                        ]
+                    }
+
+        breads = path_up_to(len(trimmed_crumbs), trimmed_crumbs)
+        return (breads, final_crumb)
+
+    def lbrowse_override(str):
+        """
+        Reset name of link text for list browse to be 'All Maps'
+
+        Args:
+            Link text
+
+        Returns:
+            New link text: 'All Maps'
+        """
+        return "All Maps"
```

Comment:
Yeah, this was sort of a hacky way to implement a last-minute change.  Originally, we had two browses for the Social Scientists Map Chicago collection: a cluster browse by subject and a list browse by date.  During our last digital collections meeting, someone asked to add a new cluster browse by date, which would group the collection items by decade.  But then we ended up with two browses called 'Date'!  One of which was really a cluster browse grouping the items by decade, and the other of which was a list browse (i.e. listing of the entire collection, ordered by date).  The link text for the browses is generated from the names they're given on the IIIF side, so rather than have:

Browse by Subject | Decade | Date

We went with:

Browse by Subject | Decade | All Maps

You're absolutely right, though, that there shouldn't be anything collection-specific in this code.  The first option that springs to mind is that we could create an optional 'link name override' for both the cluster browse and list browse items in the Wagtail admin interface.  This function will then just always get applied, but it'll default to the identity function if the field is empty, and be a constant function to the override text if the field is non-empty.  Does that sound good to you?